src/Interpreters/executeQuery.cpp

```diff
@@ -345,10 +350,39 @@ QueryLogElement logQueryStart(
 
         if (elem.type >= settings.log_queries_min_type && !settings.log_queries_min_query_duration_ms.totalMilliseconds())
         {
-            if (auto query_log = context->getQueryLog())
-                query_log->add(elem);
+            if (!settings.log_query_settings && settings.log_query_settings.changed)
+                LOG_DEBUG(
```

Comment:
So in l. 359, we call `query_log->add(elem);` (aka. write something into the query_log) even if setting `log_query_settings` is disabled. This seems intended. I guess the setting will be respected further down in the called method. But then the question is why all this new tracing is in executeQuery.cpp and not closer or next to the place where the setting is respected. Doing so will also avoid the duplicated logging further down in this file.

Reply:
Sorry, I don't get it. `log_query_settings` is about query_log.Setting field, - will be empty or with changed settings for the query. This message is for the case when user set specific some settings for a query but also don't disable logging this settings in query_log by setting `log_query_settings` to zero